UE/Tests/Application/ApplicationTestSuite.cpp

```diff
@@ -125,4 +125,30 @@ TEST_F(ApplicationConnectedTestSuite, shallHandleSms)
     objectUnderTest.handleSms(sms);
 }
 
+TEST_F(ApplicationConnectedTestSuite, shallHandleShowSmsList)
+{
+    using SmsMessages = std::vector<std::pair<Sms, SmsState>>;
+    SmsMessages messages;
+    Sms sms{PHONE_NUMBER, "example sms message"};
+    std::pair<Sms, SmsState> pair{sms,SmsState::NotViewed};
+    messages.push_back(pair);
+
+    EXPECT_CALL(smsDbMock, getSmsMessages()).WillOnce(ReturnRef(messages));
+    EXPECT_CALL(userPortMock, viewSmsList(messages));
+
+    objectUnderTest.handleShowSmsList();
+}
+
+TEST_F(ApplicationConnectedTestSuite, shallHandleShowSms)
+{
+    const unsigned indexOfSms = 0;
+    Sms sms{PHONE_NUMBER, "example sms message"};
+
+    EXPECT_CALL(smsDbMock, updateSmsState(indexOfSms));
+    EXPECT_CALL(smsDbMock, getSms(indexOfSms)).WillOnce(ReturnRef(sms));
```

Comment:
Is it necessary to declare the indexOfSms variable?
I think you can write it like this.
```c++
    EXPECT_CALL(smsDbMock, updateSmsState(_));
    EXPECT_CALL(smsDbMock, getSms(_)).WillOnce(ReturnRef(sms));
```

Reply:
That is true, but I still need indexOfSms becouse it is given for handleshowSms function and i need it to call handleshowSms. I can't write
`objectUnderTest.handleShowSms(_);`
It still must be 
`objectUnderTest.handleShowSms(indexOfSms);`